src/aa_remove_data/pb_utils.py

```diff
@@ -139,49 +182,75 @@ def read_pb(self, filepath: PathLike):
             filepath (PathLike): Path to PB file.
         """
         with open(filepath, "rb") as f:
-            first_line = self._restore_newline_chars(f.readline().strip())
-            lines = f.readlines()
-        self.header.ParseFromString(first_line)
-        proto_class = self.get_proto_class()
-        self.samples = [proto_class() for _ in range(len(lines))]
-        for i, sample in enumerate(self.samples):
-            line = self._restore_newline_chars(lines[i].strip())
-            sample.ParseFromString(line)
+            if self._start_line == 0:
+                result = subprocess.run(
+                    ["wc", "-l", filepath], stdout=subprocess.PIPE, text=True
+                )
+                self._total_lines = int(result.stdout.split()[0])
+                first_line = self._restore_newline_chars(f.readline().strip())
+                self.header.ParseFromString(first_line)
+                f.seek(0)
+                self._start_line += 1
+            end_line = min(self._start_line + self._chunk_size, self._total_lines)
```

Comment:
Similarly, a newline separation here between writing the header, calculating the next line and writing the file body may improve readability

Reply:
You may also consider extracting where you write the header into a small utility function. It may help to reduce the length and improve the readability

(e.g)
if self._start_line == 0:
  write_header()
...